sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormPage.cs

```diff
@@ -12,23 +11,24 @@ public class FormPage
     {
         internal FormPage(PageResult_internal pageResult, ReadResult_internal readResult)
         {
+            PageNumber = readResult.Page;
             TextAngle = readResult.Angle;
             Width = readResult.Width;
             Height = readResult.Height;
             Unit = readResult.Unit;
-
-            if (readResult.Lines != null)
-            {
-                Lines = ConvertLines(readResult.Lines, readResult.Page);
-            }
-
-            if (pageResult != null)
-            {
-                //TODO: set tables to empty list
-                Tables = ConvertTables(pageResult, readResult);
-            }
+            Lines = readResult.Lines != null
```

Comment:
you could avoid checking here if the type is null and just directly call `ConvertLines`. if `readResult.Lines` ends up being null or empty, the `for` won't execute and it will return an empty `List<FormLine>`.

Reply:
Unfortunately, I don't think that's possible. It seems [foreach throws a NullReferenceException when iterating over a null reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/foreach-in).

I wasn't sure whether I should put the null verification in the constructor or in the method, but it felt more like a constructor's responsibility, while the method should be responsible for parsing the arguments that are already verified. I think it's just a matter of preference at this point, but let me know if you feel strongly about it.

Reply:
I'll merge the PR to get it closed, but ping me if you have any concerns and I can fix it in the next one.